File: web/Models/ResultID.cs
using System;
using System.Text;

namespace Kumo.Models
{
    public class ResultID
    {
        private string id;

        private ResultID() { }

        public ResultID(string id)
        {
            this.id = id;
        }
        public ResultID(string url, string selector)
        {
            this.id = Convert.ToBase64String(Encoding.UTF8.GetBytes($"{url},{selector}"));
        }

        public override string ToString()
        {
            return id;
        }

        public (string Url, string Selector) Extract()
        {
            string[] tokens = Encoding.UTF8.GetString(Convert.FromBase64String(this.id)).Split(',');
            return (tokens[0], tokens[1]);
        }
    }
}

File: server/model/result_id.go
package model

import (
	"strings"
	 "encoding/base64"
)

type ResultID struct {
	id string
}

func NewResultID(URL, selector string) *ResultID {
	return &ResultID{id: base64.StdEncoding.EncodeToString([]byte(URL + "," + selector))}
}

func (r *ResultID) FromString(id string) *ResultID {
	return &ResultID{id: id}
}

func (r *ResultID) String() string {
	return r.id
}

func (r *ResultID) Extract() (*string, *string) {
	b, err := base64.StdEncoding.DecodeString(r.id)

	if err != nil {
		return nil, nil
	}

	tokens := strings.Split(",", string(b))

	return &tokens[0], &tokens[1]
}
